Build failure: materialgram & telegram-desktop on darwin

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

nix build .#telegram-desktop

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

_No response_

### Relevant log output

```console
❯ nix build .#telegram-desktop
error:
       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:37:12:
           36|
           37|   strict = derivationStrict drvAttrs;
             |            ^
           38|

       … while evaluating derivation 'telegram-desktop-6.2.3'
         whose name attribute is located at /nix/store/zc6343k4hxmjvjw9230izgc6kfpz7rsf-source/pkgs/stdenv/generic/make-derivation.nix:544:13

       … while evaluating attribute 'buildInputs' of derivation 'telegram-desktop-6.2.3'
         at /nix/store/zc6343k4hxmjvjw9230izgc6kfpz7rsf-source/pkgs/stdenv/generic/make-derivation.nix:596:13:
          595|             depsHostHost = elemAt (elemAt dependencies 1) 0;
          596|             buildInputs = elemAt (elemAt dependencies 1) 1;
             |             ^
          597|             depsTargetTarget = elemAt (elemAt dependencies 2) 0;

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: Package ‘wayland-1.24.0’ in /nix/store/zc6343k4hxmjvjw9230izgc6kfpz7rsf-source/pkgs/development/libraries/wayland/default.nix:98 is marked as broken, refusing to evaluate.

       a) To temporarily allow broken packages, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_BROKEN=1

          Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
                then pass `--impure` in order to allow use of environment variables.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowBroken = true; }
       in configuration.nix to override this.

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowBroken = true; }
       to ~/.config/nixpkgs/config.nix.
```

### Additional context
last working nixpkgs-unstable cb82756ecc37fa623f8cf3e88854f9bf7f64af93

### System metadata

 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.6.0, macOS 15.7.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.31.2`
 - nixpkgs: `/nix/store/pv11fms46b44vv50nvi3l1ckjdlrm6lf-source`


### Notify maintainers

@NickCao @oluceps @Aleksanaa @stellessia

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

@alapshin

after fixing withWayland, there is another issue. https://github.com/NixOS/nixpkgs/pull/455171#issuecomment-3441944338 
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
FWNetwork
    linked by target "Network" in directory /nix/store/p8dkm0aadr4qjs825gk8fzkfspx84lid-qtbase-6.10.0/.build/qtbase-everywhere-src-6.10.0/src/network
    linked by target "QAppleNetworkInformationPlugin" in directory /nix/store/p8dkm0aadr4qjs825gk8fzkfspx84lid-qtbase-6.10.0/.build/qtbase-everywhere-src-6.10.0/src/plugins/networkinformatio>
```

--- a/pkgs/development/libraries/wayland/default.nix
+++ b/pkgs/development/libraries/wayland/default.nix
@@ -107,7 +107,8 @@ stdenv.mkDerivation (finalAttrs: {
     homepage = "https://wayland.freedesktop.org/";
     license = licenses.mit; # Expat version
     platforms = platforms.unix;
-    broken = stdenv.hostPlatform.isDarwin; # requires more work: https://gitlab.freedesktop.org/wayland/wayland/-/merge_requests/481
+    # requires more work: https://gitlab.freedesktop.org/wayland/wayland/-/merge_requests/481
+    badPlatforms = lib.platforms.darwin;
     maintainers = with maintainers; [
       codyopel
       qyliss

